src/functions.js

```diff
@@ -0,0 +1,127 @@
+import { updateLikes } from './likes.js';
+import { createComment, getComment } from './comment.js';
+import { countComments } from './counts.js';
+
+const mealsURL = 'https://www.themealdb.com/api/json/v1/1/filter.php?c=Seafood';
+const mealsContainer = document.querySelector('.meals-container');
+
+const getMealInstr = async (id) => {
+  const url = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${id}`;
+  const response = await fetch(url);
+  const data = await response.json();
+  const inst = data.meals[0].strInstructions;
+  return inst;
+};
+
+const getData = async () => {
+  const response = await fetch(mealsURL);
+  const data = await response.json();
+  return data;
+};
+
+const displayMeals = async () => {
+  mealsContainer.innerHTML = '';
+  const mealsOb = await getData();
+  mealsOb.meals.forEach((meal) => {
+    const mealDiv = document.createElement('div');
+    mealDiv.classList.add('meal-div');
+    // Call updateLikes() function to update the like count of the meal
+    updateLikes(meal, mealDiv);
+
+    mealDiv.innerHTML = `<div class="img-div" id=${meal.idMeal}>
+    <img src=${meal.strMealThumb} alt=${meal.strMeal} class="meal-img">
+  </div>
+  <a href="#" class="meal-title" data-id=${meal.idMeal}>${meal.strMeal}</a>
+  <div class="meals-details">
+  <i class="far fa-heart like-btn"></i>
+  <p class="like-count">Likes loading</p>
+  </div>
+  <button class="comment-btn" id="${meal.idMeal}">Comments</button>`;
+
+    // Append the list item element to the unordered list element
+    mealsContainer.appendChild(mealDiv);
+    // create the popup for comments
+
+    const commentButton = mealDiv.querySelector('.comment-btn');
+    commentButton.addEventListener('click', async (event) => {
+      event.preventDefault();
+      const mealId = event.target.id;
+      console.log(mealId);
+      // get meal instructions
+      const instructions = await getMealInstr(mealId);
+      console.log(typeof (instructions));
+      // const comLen = await countComments();
+
+      const popupContainer = document.createElement('div');
+      popupContainer.classList.add('popup-container');
+
+      popupContainer.id = 'popup-container';
+
+      const popupContent = `
+        <button class="close">X</button>
+        <h2>Meal Instructions</h2>
+        <p id="meal-instructions">${instructions}</p>
+        <h2>Comments (<span id="comment-counter"></span>)</h2>
+        <ul id="comments-list"></ul>
+        <h2>Add Comment</h2>
```

Comment:
- [ ] I think you should consider following the mockup design of the application. For this case, I suggest to also include the image and title of the meal. This improves the user-interface of the application as the user will be confident that the popup is on that particular meal clicked on and going to be commented about ☺️.

Reply:
Well resolved ✔️

![image](https://user-images.githubusercontent.com/38088818/233781319-57a8dfbf-c4dd-4fcf-92ad-05564af2848f.png)

Reply:
Thanks, @KDlamini. Does that mean the project is approved? I'm still seeing *review pending* in the dashboard.